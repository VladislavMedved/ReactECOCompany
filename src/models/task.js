import { types } from "mobx-state-tree";

export const Task = types.model({
    taskName: types.string,
    completed: false,
    assignee: types.maybeNull(types.string)
}).actions(self => ({
    setAssignee(assignee) {
        self.assignee = assignee;
    },
    complete(completed) {
        self.completed = completed;
    }
}));