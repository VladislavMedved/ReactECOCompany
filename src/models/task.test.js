import { getSnapshot, onSnapshot, onPatch } from "mobx-state-tree";
import { Task } from "./task";

describe("Mobx-State-tree models", () => {

    const taskSnapshot = {
        taskName: "fix 3ds FIS functionality on ClientDetails"
    };

    it("should be possible to complete the task", () => {
        var task = Task.create(taskSnapshot);
        expect(task.completed).toBe(false);
        task.complete(true);
        expect(task.completed).toBe(true);
    });

    it("should be possible to set assignee to the task", () => {
        var task = Task.create(taskSnapshot);


        var patches = [];
        onPatch(task, patch => {
            patches.push(patch);
        })

        expect(task.assignee).toBe(null);
        task.setAssignee("Igor");
        task.setAssignee("Igor Schuchenko")
        expect(task.assignee).toBe("Igor Schuchenko");

        expect(task).toMatchSnapshot();

        expect(patches).toMatchSnapshot();
    });
})