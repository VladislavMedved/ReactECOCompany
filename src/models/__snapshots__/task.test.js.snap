// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobx-State-tree models should be possible to set assignee to the task 1`] = `
Object {
  "assignee": "Igor Schuchenko",
  "completed": false,
  "taskName": "fix 3ds FIS functionality on ClientDetails",
}
`;

exports[`Mobx-State-tree models should be possible to set assignee to the task 2`] = `
Array [
  Object {
    "op": "replace",
    "path": "/assignee",
    "value": "Igor",
  },
  Object {
    "op": "replace",
    "path": "/assignee",
    "value": "Igor Schuchenko",
  },
]
`;
